.github/workflows/puma.yml

```diff
@@ -19,15 +19,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu, macos, windows ]
-        ruby: [ 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, head, jruby, truffleruby-head ]
+        os: [ ubuntu-20.04, macos-10.15, windows-2019 ]
```

Comment:
I see the value in doing some separate Ubuntu 20/18 testing, but why pin our mac and windows tests too?

Reply:
I don't have strong feeling about it.

Right now, ubuntu-latest is 18.04.  I don't like pinning Ruby versions, but I think OS's is reasonable, and for a person unfamiliar with Actions, the numeric callout is clear?

I can add some MRI Ruby to ubuntu-18.04.  Thoughts?

Reply:
It's just kinda a weird situation where ubuntu "latest" is not 20.04. 

I think we just test the latest MRI Ruby version on 20.04. That seems like enough coverage IMO.

Reply:
> It's just kinda a weird situation where ubuntu "latest" is not 20.04.

I have a lot of Actions bookmarks, but somewhere they have listings of dates re runner OS's. etc.  I'll look for it (later), and add it to the workflow.  IOW, I think it's where they'd announce that 20.04 will become 'latest'.  I wouldn't be surprised if they deprecate the 'latest' concept.  Note that every change here was needed to get 20.04 CI to pass...

Maybe test MRI head, 2.7 & 2.5 on 20.04, and run all the tests on 18.04, as was done previously?

Or, what would you like?  I'm working on #2337...

Reply:
> I have a lot of Actions bookmarks

Me too :) 

https://docs.github.com/en/actions/reference/software-installed-on-github-hosted-runners says "Note: The Ubuntu 20.04 virtual environment is currently provided as a preview only." (https://github.com/actions/virtual-environments#available-environments says the same thing)

According to https://github.com/actions/virtual-environments/issues/1430 there doesn't seem to be any date set when the preview period will be over.

Reply:
I may have been mistaken, but they often pin issues (https://github.com/actions/virtual-environments/issues), I think that was the case when they thought they'd remove windows-2016, which people complained about...